src/uct/api/uct.h

```diff
@@ -106,6 +106,90 @@ BEGIN_C_DECLS
  * @}
  */
 
+/**
+ * @defgroup UCT_CLIENT_SERVER   UCT client-server operations
+ * @ingroup UCT_API
+ * @{
+ * Defines client-server operations.
+ * The client-server API allows the connection establishment between an active
+ * side - a client, and its peer - the passive side - a server.
+ * The connection can be established through a UCT transport that supports
+ * listening and connecting via IP address and port (listening can also be on INADDR_ANY).
+ *
+ * The following is a general overview of the operations on the server side:
+ *
+ * Connecting:
+ * @ref uct_cm_open
+ *      Open a connection manager.
+ * @ref uct_listener_create
+ *      Create a listener on the CM and start listening on a given IP,port / INADDR_ANY.
+ * @ref uct_listener_conn_request_callback_t
+ *      This callback is invoked by the UCT transport to handle an incoming connection
+ *      request from a client.
+ *      Accept or reject the client's connection request.
+ * @ref uct_ep_create
+ *      Connect to the client by creating an endpoint in case of accepting its request.
+ * @ref uct_sockaddr_priv_pack_callback_t
+ *      This callback is invoked by the UCT transport to fill auxiliary data in
+ *      the connection acknowledgement or reject notification back to the client.
+ *      Send the client a connection acknowledgement or reject notification.
+ *      Wait for an acknowledgment from the client, indicating that it is connected.
+ * @ref uct_ep_server_connect_cb_t
+ *      This callback is invoked by the UCT transport to handle the connection
+ *      acknowledgment from the client.
+ *
+ * Disconnecting:
+ * @ref uct_ep_disconnect
+ *      Disconnect the server's endpoint from the client.
+ *      Can be called when initiating a disconnect or when receiving a disconnect
+ *      notification from the remote side.
+ * @ref uct_ep_disconnect_cb_t
+ *      This callback is invoked by the UCT transport when the client side calls
+ *      uct_ep_disconnect as well.
+ * @ref uct_ep_destroy
+ *      Destroy the endpoint connected to the remote peer.
```

Comment:
@shamisp calling destroy on an endpoint before completing a non-blocking operation on it will result in not finishing the operation - this is why we have the disconnect_cb, to indicate that the disconnect completed.